Source/WebCore/page/writing-tools/WritingToolsController.mm

```diff
@@ -632,6 +640,42 @@
     m_page->chrome().client().proofreadingSessionUpdateStateForSuggestionWithID(data.sessionID, WritingTools::TextSuggestion::State::Reviewing, data.suggestionID);
 }
 
+void WritingToolsController::respondToUnappliedEditing(EditCommandComposition* composition)
+{
+    for (auto& value : m_states.values()) {
+        auto* state = std::get_if<CompositionState>(&value);
+        if (!state)
+            continue;
+
+        auto appliedCommandIsWritingToolsCommand = std::ranges::any_of(state->reappliedCommands, [&](const auto& command) {
```

Comment:
Would it be possible to detect that the command is a Writing Tools command via method on `EditCommandComposition` or another command class?

I'll expect I'll need something similar.

Reply:
(Chatted offline about adding a new `EditAction` for this)